public/app/plugins/datasource/prometheus/result_transformer.ts

```diff
@@ -37,6 +41,97 @@ interface TimeAndValue {
   [TIME_SERIES_VALUE_FIELD_NAME]: number;
 }
 
+// V2 result trasnformer used to transform query results from queries that were run trough prometheus backend
+export function transformV2(response: DataQueryResponse, options: DataQueryRequest<PromQuery>) {
+  const promResults: DataFrame[] = cloneDeep(response.data);
+
+  // Get refIds that have table format as we need to process those to table reuslts
+  const tableRefIds = options.targets.map((target) => {
+    if (target.format === 'table') {
+      return target.refId;
+    }
+    return;
+  });
```

Comment:
there is one potential problem here:
- the `tableRefIds` array can contain `undefined`s
- lower there, `dataFrame.refId` can be `undefined` (based on typescript info)
- so, in theory, if you return time_series results, and they have no `refId`, they will be considered table-results. i do not know if this can happen or not.

here is a way to get the `tableRefIds` without `undefined`s, but then some more fixing will be needed lower, because then suddenly typescript complains that you cannot check if a `string|undefined` is contained in a string-array...

so i recommend changing this to not have this problem, but if you are very sure it can never happen, then it is ok.

```suggestion
  const tableRefIds = options.targets.filter((target) => target.format === 'table').map((target) => target.refId);
```

i am a little worried about the fact that `tableRefIds` does contain `undefined` items too, and below, `dataFrame.refId` can be `undefined`... so if you get returned dataframes without `refId` then those will be considered tableRefIds or something... i don't know how much this can happen.

i wrote an alternative version that does not cause `undefineds`, but then later below typescript will complain, so i don't know 😿

Reply:
I agree. Thanks for pointing it out. I've  tackled it this way, let me know what do you think https://github.com/grafana/grafana/pull/39133/commits/d7707c58ca140ec1be53907ec7fa350b035773a2

Also thanks for sharing the `partition`. I didn't know about it and it is super useful in this scenario!

Reply:
looks fine! :+1: